src/Controller/Frontend.php

```diff
@@ -254,7 +254,25 @@ public function listing(Request $request, $contenttypeslug)
             $amount = empty($contenttype['listing_records']) ? $this->getOption('general/listing_records') : $contenttype['listing_records'];
         }
         if (!$order = $this->getOption('theme/listing_sort', false)) {
-            $order = empty($contenttype['sort']) ? $this->getOption('general/listing_sort') : $contenttype['sort'];
+            $order = empty($contenttype['sort']) ? null : $contenttype['sort'];
+        }
+        // If $order is not set, one of two things can happen: Either we let `getContent()` sort by itself, or we
+        // explicitly set it to sort on the general/listing_sort setting.
+        if ($order === null) {
+            $taxonomies = $this->getOption('taxonomy');
+            $hassortorder = false;
+            if (!empty($contenttype['taxonomy'])) {
+                foreach($contenttype['taxonomy'] as $contenttypetaxonomy) {
+                    if ($taxonomies[ $contenttypetaxonomy ]['has_sortorder']) {
+                        // We have a taxonomy with a sortorder, so we must keep $order = false, in order
+                        // to let `getContent()` handle it. We skip the fallback that's a few lines below.
+                        $hassortorder = true;
```

Comment:
I just tested this here and the only way I could trigger this is due to the break being outside the `foreach` ergo inside the `if` instead.

Again, re-enforcing the Ultra Level Gawain Derp today, but do you have a simple reproducible test case I can start with?